pkgs/by-name/ma/material-maker/package.nix

```
@@ -0,0 +1,159 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  fetchurl,
+  nix-update-script,
+
+  # build deps
+  autoPatchelfHook,
+  copyDesktopItems,
+  makeDesktopItem,
+  undmg,
+
+  # runtime deps
+  libGL,
+  libGLU,
+  libX11,
+  libXcursor,
+  libXext,
+  libXfixes,
+  libXi,
+  libXinerama,
+  libXrandr,
+  libXrender,
+  alsa-lib,
+  libpulseaudio,
+  udev,
+}:
+let
+  pname = "material-maker";
+  version = "1.3";
+  versionUnderscore = "${lib.replaceStrings [ "." ] [ "_" ] version}";
+
+  sources = {
+    linux = fetchzip {
+      url = "https://github.com/RodZill4/material-maker/releases/download/${version}/material_maker_${versionUnderscore}_linux.tar.gz";
+      hash = "sha256-WEu5gVfnswB5zYzu3leOL+hKOBzJbn48gHQKshlfOh4=";
+    };
+    darwin = fetchurl {
+      url = "https://github.com/RodZill4/material-maker/releases/download/${version}/material_maker_${versionUnderscore}.dmg";
+      hash = "sha256-D4jPQEOKws2JFczh0K55mR08jIBMU6jOFOm6TyFxnt8=";
+    };
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  inherit pname version;
+
+  src =
+    if stdenv.isLinux && stdenv.isx86_64 then
+      sources.linux
+    else if stdenv.isDarwin then
+      sources.darwin
+    else
+      throw "unsupported platform";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+  ] ++ lib.optionals stdenv.isDarwin [ undmg ];
+
+  sourceRoot = ".";
+
+  buildInputs = [
+    libGL
+    libGLU
+    libX11
+    libXcursor
+    libXext
+    libXfixes
+    libXi
+    libXinerama
+    libXrandr
+    libXrender
+  ];
+
+  desktopItems = makeDesktopItem {
+    name = "material-maker";
+    exec = "material-maker";
+    icon = "material-maker";
+    genericName = "Procedural texture generation and 3D model painting tool";
+    desktopName = "Material Maker";
+    comment = finalAttrs.meta.description;
+    terminal = false;
+    categories = [
+      "Graphics"
+      "3DGraphics"
+    ];
+    keywords = [
+      "material"
+      "textures"
+      "paint"
+      "3d"
+    ];
+  };
+
+  installPhase =
+    ''
+      runHook preInstall
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      install -D -m 755 $src/material_maker.x86_64 $out/opt/material-maker
+      install -D -m 644 $src/material_maker.pck $out/opt/material-maker.pck
+
+      for dir in library nodes environments meshes export; do
+        cp -R $src/$dir $out/opt
+      done
+
+      cp -R $src/examples $examples
+
+
+      mkdir -p $out/share/
+      cp -R $src/doc $out/share
+
+      install -D -m 644 $src/doc/_static/icon.png -t $out/share/icons/hicolor/256x256/apps/
+
+      mkdir -p $out/bin
+      ln -s $out/opt/material-maker $out/bin/material-maker
+    ''
+    + lib.optionalString (builtins.elem "doc" finalAttrs.meta.outputsToInstall && stdenv.isLinux) ''
+      ln -s $doc/share/doc $out/opt/doc
+    ''
+    +
+      lib.optionalString (builtins.elem "examples" finalAttrs.meta.outputsToInstall && stdenv.isLinux)
+        ''
+          ln -s $examples $out/opt/examples
+        ''
```

```suggestion
    + lib.optionalString (builtins.elem "examples" finalAttrs.meta.outputsToInstall && stdenv.isLinux) ''
      ln -s $examples $out/opt/examples
    ''
```

Nitpick to uniform style